File: src/main/java/com/DavidM1A2/othertest/common/constants/ModSounds.java
package com.DavidM1A2.afraidofthedark.common.constants;

import net.minecraft.util.ResourceLocation;
import net.minecraft.util.SoundEvent;

/**
 * A class containing all sounds in AOTD
 */
public class ModSounds
{
    public static final SoundEvent CROSSBOW_FIRE = create("crossbow_fire");
    public static final SoundEvent CROSSBOW_LOAD = create("crossbow_load");
    public static final SoundEvent WEREWOLF_IDLE = create("werewolf_idle");
    public static final SoundEvent WEREWOLF_AGRO = create("werewolf_agro");
    public static final SoundEvent WEREWOLF_DEATH = create("werewolf_death");
    public static final SoundEvent WEREWOLF_HURT = create("werewolf_hurt");
    public static final SoundEvent JOURNAL_SIGN = create("journal_sign");
    public static final SoundEvent ACHIEVEMENT_UNLOCKED = create("achievement_unlocked");
    public static final SoundEvent PAGE_TURN = create("page_turn");
    public static final SoundEvent BUTTON_HOVER = create("button_hover");
    public static final SoundEvent SPELL_CRAFTING_BUTTON_HOVER = create("spell_crafting_button_hover");
    public static final SoundEvent BELLS = create("bells");
    public static final SoundEvent ERIE_ECHOS = create("erie_echos");
    public static final SoundEvent KEY_TYPED = create("key_typed");

    // An array containing a list of sounds that AOTD adds
    public static final SoundEvent[] SOUND_LIST = new SoundEvent[]
            {
                    CROSSBOW_FIRE,
                    CROSSBOW_LOAD,
                    WEREWOLF_IDLE,
                    WEREWOLF_AGRO,
                    WEREWOLF_DEATH,
                    WEREWOLF_HURT,
                    JOURNAL_SIGN,
                    ACHIEVEMENT_UNLOCKED,
                    PAGE_TURN,
                    BUTTON_HOVER,
                    SPELL_CRAFTING_BUTTON_HOVER,
                    BELLS,
                    ERIE_ECHOS,
                    KEY_TYPED
            };

    /**
     * Utility function to create sound events from the sounds.json file
     *
     * @param name The name of the sound
     * @return A sound event representing that sound
     */
    private static SoundEvent create(String name)
    {
        ResourceLocation location = new ResourceLocation(Constants.MOD_ID, name);
        return new SoundEvent(location).setRegistryName(location);
    }
}

File: src/main/java/com/davidm1a2/afraidofthedark/common/constants/ModSounds.kt
package com.davidm1a2.afraidofthedark.common.constants

import net.minecraft.util.ResourceLocation
import net.minecraft.util.SoundEvent

/**
 * A class containing all sounds in AOTD
 */
object ModSounds {
    val CROSSBOW_FIRE = create("crossbow_fire")
    val CROSSBOW_LOAD = create("crossbow_load")
    val WEREWOLF_IDLE = create("werewolf_idle")
    val WEREWOLF_AGRO = create("werewolf_agro")
    val WEREWOLF_DEATH = create("werewolf_death")
    val WEREWOLF_HURT = create("werewolf_hurt")
    val ENCHANTED_FROG_CROAK = create("enchanted_frog_croak")
    val ENCHANTED_FROG_DEATH = create("enchanted_frog_death")
    val ENCHANTED_FROG_HURT = create("enchanted_frog_hurt")
    val JOURNAL_SIGN = create("journal_sign")
    val ACHIEVEMENT_UNLOCKED = create("achievement_unlocked")
    val PAGE_TURN = create("page_turn")
    val BUTTON_HOVER = create("button_hover")
    val SPELL_CRAFTING_BUTTON_HOVER = create("spell_crafting_button_hover")
    val BELLS = create("bells")
    val EERIE_ECHOS = create("eerie_echos")
    val KEY_TYPED = create("key_typed")
    val SPELL_CAST = create("spell_cast")
    val NIGHTMARE_MUSIC = create("nightmare_music")
    val NIGHTMARE_CHASE_MUSIC = create("nightmare_chase_music")

    // An array containing a list of sounds that AOTD adds
    val SOUND_LIST = arrayOf(
        CROSSBOW_FIRE,
        CROSSBOW_LOAD,
        WEREWOLF_IDLE,
        WEREWOLF_AGRO,
        WEREWOLF_DEATH,
        WEREWOLF_HURT,
        ENCHANTED_FROG_CROAK,
        JOURNAL_SIGN,
        ACHIEVEMENT_UNLOCKED,
        PAGE_TURN,
        BUTTON_HOVER,
        SPELL_CRAFTING_BUTTON_HOVER,
        BELLS,
        EERIE_ECHOS,
        KEY_TYPED,
        SPELL_CAST,
        NIGHTMARE_MUSIC,
        NIGHTMARE_CHASE_MUSIC
    )

    /**
     * Utility function to create sound events from the sounds.json file
     *
     * @param name The name of the sound
     * @return A sound event representing that sound
     */
    private fun create(name: String): SoundEvent {
        val location = ResourceLocation(Constants.MOD_ID, name)
        return SoundEvent(location).setRegistryName(location)
    }
}